### PR TITLE
Go 2 UI

### PR DESCRIPTION
**Pull request by stephdl**
Write UI for crowsdesc

image
image
image
image

### CODE REVIEW

File: ui/public/i18n/en/translation.json [hunk 1]
```diff
@@ -24,7 +24,32 @@
     "title": "Settings",
     "configure_instance": "Configure {instance}",
     "save": "Save",
-    "test_field": "Test field"
+    "receiver_emails": "Email notifications",
+    "receiver_emails_list": "List of email recipients for notifications",
+    "whitelists": "Allow list of CIDR, IP and Fully qualified domain name",
```
Comment (gsanchietti) [inline ui/public/i18n/en/translation.json:29]:
```suggestion
    "whitelists": "Allow list of CIDR, IP and fully qualified domain name",
```

File: ui/public/i18n/en/translation.json [hunk 2]
```diff
@@ -24,7 +24,32 @@
...
+    "disabled": "Disabled",
+    "dyn_bantime": "Dynamic ban time",
+    "dyn_bantime_tips": "Ban time is calculated by ((number of attempts + 1) * 4) minutes",
+    "bantime": "Bantime",
+    "minutes": "Minutes",
+    "disable_online_api": "Disable central API",
+    "disable_online_api_tips": "The central API is used to retrieve blacklists and to have a glance on metrics at app.crowdsec.net",
+    "open_app_crowdsec": "App crowdsec website",
+    "enroll_instance": "Enroll the crowdsec instance",
+    "enroll_instance_must_be_real_token": "The token must be retrieved on the website app.crowdsec.net",
```
Comment (gsanchietti) [inline ui/public/i18n/en/translation.json:45]: could you try embedding a URL, I think it should work
```suggestion
    "enroll_instance_must_be_real_token": "The token must be retrieved on the website <a href='app.crowdsec.net'>app.crowdsec.net</a>",
```

File: ui/public/i18n/en/translation.json [hunk 3]
```diff
@@ -24,7 +24,32 @@
...
+    "enroll_instance_placeholder": "cl7qa8dsfdsf00030vl7hguyiog",
+    "helo_host": "Helo_host",
+    "helo_host_must_be_relevant_for_smtp": "You can set a specific Helo FQDN if your antispam add score due to default localhost Helo name",
```
Comment (gsanchietti) [inline ui/public/i18n/en/translation.json:48]:
```suggestion
    "helo_host_must_be_relevant_for_smtp": "You can set a specific Helo FQDN if your antispam adds score due to default localhost Helo name",
```

File: ui/public/i18n/en/translation.json (hunk 2 above)
Comment (stephdl) [reply ui/public/i18n/en/translation.json:45]: I do not like to add html inside translations, we will have error with translators one day, I added a button to reach the website, does it sound enough ?

Comment (stephdl) [reply ui/public/i18n/en/translation.json:45]: image

Comment (gsanchietti) [reply ui/public/i18n/en/translation.json:45]: Thanks, for me it's ok.

**Review by gsanchietti — APPROVED** [review]

File: ui/public/i18n/en/translation.json [hunk 4]
```diff
@@ -24,7 +24,36 @@
...
+    "enroll_instance_placeholder": "xxxxxxxxxxxxxxxxxxx",
+    "helo_host": "Helo_host",
+    "helo_host_must_be_relevant_for_smtp": "You can set a specific Helo FQDN if your antispam adds score due to default localhost Helo name",
+    "helo_host_placeholder": "sub.domain.com",
+    "enable_smarthosts_for_notifications": "Smarthost settings",
+    "bad_email_address": "Bad email address",
+    "bad_IP_or_hostname": "Bad CIDR, IP or FQDN",
+    "processing": "Processing...",
+    "required": "Required",
+    "token_not_valid":"The attachment key provided is not valid",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:55]: Let's always call it "token"
```suggestion
    "token_not_valid":"The token provided is not valid",
```

File: README.md [hunk 5]
```diff
@@ -46,19 +46,27 @@ You can also modify settings with the configure-module action
...
     }
     EOF
 
-- `bantime`: set the ban time `m` for minute, `h` for hours
+- `bantime`: set the ban time in minute
 - `dyn_bantime`: enable a dynamic ban_time ((number of ban +1) *4) (same unit as ban_time)
 - `receiver_emails`: all emails account you want to notice when decisions or alert come
 - `helo_host`: set a fully qualified domain name to use the relevant helo with postfix.(could be empty `""`)
 - `whitelists`: whitelist domain, ip or network to crowdsec, no ban will occurs for that list
 - `disable_online_api`: enable/disable to  push signals and receive bad IPs from crowdsec hub (true/false default)
```
[the remark this answers is not in the document]
Comment (stephdl) [reply README.md:62]: you are right hum https://github.com/NethServer/ns8-crowdsec/blob/43f8a16915f0b55c42c3495fc17cc4f03a0af06b/imageroot/actions/get-configuration/20read#L24

[the remark this answers is not in the document]
Comment (stephdl) [reply README.md:62]: I would like to have central api enabled by default I think, I need the input of @gsanchietti 

Central API gives you the availability to manage block list and monitor the bans, what do you think @andre8244 and @gsanchietti central api is enabled or disabled by default, I vote enabled

[the remark this answers is not in the document]
Comment (gsanchietti) [reply README.md:62]: I'd go with enabled: it gives a better protection.
Anyone can opt-out.

[the remark this answers is not in the document]
Comment (andre8244) [reply README.md:62]: Enabled by default is ok for me. If possible I would rename the variable to `enable_online_api` to make it more clear

[the remark this answers is not in the document]
Comment (stephdl) [reply README.md]: @andre8244 could you review I inversed the logic https://github.com/NethServer/ns8-crowdsec/pull/13/commits/b83150efa83c9bf7643870a0755c917031b5cb5a

File: ui/src/store/index.js [hunk 6]
```diff
@@ -9,7 +9,7 @@ Vue.use(Vuex);
 
 export default new Vuex.Store({
   state: {
-    appName: "",
+    appName: "Crowdsec",
```
Comment (andre8244) [inline ui/src/store/index.js:12]:
```suggestion
    appName: "CrowdSec",
```

**Review by andre8244 — COMMENTED** [review]
Just a typo in `appName`, then we can merge

**Review by andre8244 — APPROVED** [review]

**Pull request merged**